reward/staking_manager.go

```diff
@@ -109,36 +142,46 @@ func (sm *StakingManager) updateStakingInfo(blockNum uint64) (*StakingInfo, erro
 }
 
 // Subscribe setups a channel to listen chain head event and starts a goroutine to update staking cache.
-func (sm *StakingManager) Subscribe() {
-	sm.chainHeadSub = sm.blockchain.SubscribeChainHeadEvent(sm.chainHeadChan)
+func SubscribeStakingManager() {
+	if stakingManager == nil {
+		logger.Debug("unable to subscribe; stakingManager is not set")
```

Comment:
How about increasing the log level to `WARN` or `ERROR`? 
It could call `NewStakingManager` instead of the logger.

Reply:
I changed to `WARN` and added more information.